### PR TITLE
Feature/better masonry

### PR DESCRIPTION
**Pull request by joshmarom**

### CODE REVIEW

File: src/js/types/masonry.js [hunk 1]
```diff
@@ -11,49 +11,56 @@ export default class Masonry extends BaseGalleryType {
...
-				imageData = this.getImageData( index ),
+			const imageData = this.getImageData( index ),
 				itemHeight = itemWidth / imageData.ratio;
+			let indexAtRow = index % columns;
 
-			item.style.setProperty( '--item-height', ( imageData.height / imageData.width * 100 ) + '%' );
-			item.style.setProperty( '--column', indexAtRow );
+			columnHeight = heights[ indexAtRow ];
+			jQuery.each( heights, ( colNumber, colHeight ) => {
+				if ( ( colHeight > 5 ) && columnHeight > colHeight + 5 ) {
```
Comment (kobizz) [inline src/js/types/masonry.js]:
```suggestion
				if ( colHeight > 5 && columnHeight > colHeight + 5 ) {
```

File: src/js/types/masonry.js [hunk 2]
```diff
@@ -11,49 +11,56 @@ export default class Masonry extends BaseGalleryType {
...
-			rows = Math.floor( this.settings.items.length / columns ),
-			rowsRemainder = this.settings.items.length % columns,
-			highestColumnsGapsCount = rowsRemainder > highestColumnIndex ? rows : rows - 1,
+			rows = itemsInColumn[ highestColumnIndex ],
+			highestColumnsGapsCount = rows - 1,
 			containerAspectRatio = highestColumn / containerWidth;
 
 		this.$container[ 0 ].style.setProperty( '--columns', columns );
-		this.$container[ 0 ].style.setProperty( '--highest-column-gap-count', highestColumnsGapsCount );
+		this.$container[ 0 ].style.setProperty( '--highest-column-gap-count', highestColumnsGapsCount.toString() );
```
Comment (kobizz) [inline src/js/types/masonry.js]: Why did you need to convert it to a string?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: src/js/types/masonry.js (hunk 2 above)
Comment (joshmarom) [reply src/js/types/masonry.js]: PhpStorm claims this function expects a string... 
Should I ignore it?

Comment (kobizz) [reply src/js/types/masonry.js]: Yes, you should. It's unnecessary.

File: src/js/types/masonry.js [hunk 3]
```diff
@@ -11,36 +11,46 @@ export default class Masonry extends BaseGalleryType {
...
 
 		const heights = [],
+			itemsInColumn = [],
 			aggregatedHeights = [],
 			columns = this.getCurrentDeviceSetting( 'columns' ),
 			containerWidth = this.$container.width(),
 			horizontalGap = this.getCurrentDeviceSetting( 'horizontalGap' ),
 			itemWidth = ( containerWidth - ( horizontalGap * ( columns - 1 ) ) ) / columns,
 			$items = this.getActiveItems();
+		let naturalColumnHeight = 0;
```
Comment (kobizz) [inline src/js/types/masonry.js:22]: Add a line break after `const` statement

File: src/js/types/masonry.js [hunk 4]
```diff
@@ -11,36 +11,46 @@ export default class Masonry extends BaseGalleryType {
...
+			heights[ i ] = 0;
+		}
 
 		$items.each( ( index, item ) => {
-			const row = Math.floor( index / columns ),
-				indexAtRow = index % columns,
-				imageData = this.getImageData( index ),
+			const imageData = this.getImageData( index ),
 				itemHeight = itemWidth / imageData.ratio;
+			let indexAtRow = index % columns;
```
Comment (kobizz) [inline src/js/types/masonry.js:33]: Add a line break after `const` statement

File: src/js/types/masonry.js [hunk 5]
```diff
@@ -49,11 +59,8 @@ export default class Masonry extends BaseGalleryType {
 		this.$container.css( 'padding-bottom', ( containerAspectRatio * 100 ) + '%' );
 
 		$items.each( ( index, item ) => {
-			const percentHeight = aggregatedHeights[ index ] ? aggregatedHeights[ index ] / highestColumn * 100 : 0,
-				row = Math.floor( index / columns );
-
+			const percentHeight = aggregatedHeights[ index ] ? aggregatedHeights[ index ] / highestColumn * 100 : 0;
```
Comment (kobizz) [inline src/js/types/masonry.js:64]: Add a line break after `const` statement

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]
